Firebase.cpp

```diff
@@ -15,100 +15,158 @@
 //
 #include "Firebase.h"
 
-const char* firebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
-const uint16_t firebasePort = 443;
+namespace {
+const char* kFirebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
+const uint16_t kFirebasePort = 443;
+}  // namespace
 
-Firebase::Firebase(const String& host) : _host(host) {
-  _http.setReuse(true);
+Firebase::Firebase(const String& host) : connection_(host) {
 }
 
 Firebase& Firebase::auth(const String& auth) {
-  _auth = auth;
+  connection_.auth(auth);
   return *this;
 }
 
-String Firebase::get(const String& path) {
-  return sendRequest("GET", path);
+FirebaseResult Firebase::get(const String& path) {
+  return connection_.sendRequestGetBody("GET", path);
 }
 
-String Firebase::push(const String& path, const String& value) {
-  return sendRequest("POST", path, value);
+FirebaseResult Firebase::push(const String& path, const String& value) {
+  return connection_.sendRequestGetBody("POST", path, value);
 }
 
-Firebase& Firebase::stream(const String& path) {
-  _error.reset();
-  String url = makeURL(path); 
+FirebaseResult Firebase::remove(const String& path) {
+  return connection_.sendRequest("DELETE", path);
+}
+
+/* FirebaseEventStream */
+
+FirebaseEventStream::FirebaseEventStream(const String& host) : connection_(host) {
+}
+
+FirebaseEventStream& FirebaseEventStream::auth(const String& auth) {
+  connection_.auth(auth);
+  return *this;
+}
+
+FirebaseResult FirebaseEventStream::connect(const String& path) {
```

Comment:
what do you think of moving this to the connection class, that would allow you to get rid of the `httpClient` and `makeURL` accessors.